base/special/gamma.jl

```diff
@@ -482,3 +482,7 @@ eta(x::Integer) = eta(Float64(x))
 eta(x::Real)    = oftype(float(x),eta(Float64(x)))
 eta(z::Complex) = oftype(float(z),eta(Complex128(z)))
 @vectorize_1arg Number eta
+
+Base.binomial{S<:Number,T<:Number}(n::S, k::T) = ( (isinteger(n) && isinteger(k))
```

Comment:
What about converting always to `BigInt` instead of `Int64`? 

```? convert(promote_type(T,S), binomial(round(BigInt, n), round(BigInt, k)))```

it would be a little slower but will address the corner case of those crazy enough to evaluate binomials with such big numbers